tests/Utils/UtilsTest.php

```diff
@@ -348,8 +349,28 @@ public function categoryProvider(): array
      */
     public function testCategoryPathIsProperlyBuilt(array $breadCrumbs, string $expectedCategoryPath): void
     {
-        $categoryPath = Utils::buildCategoryPath($breadCrumbs);
+        $navigationCategoryMock = $this->getMockBuilder(CategoryEntity::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $navigationCategoryMock->expects($this->once())->method('getBreadcrumb')->willReturn(['Main']);
+
+        $categoryPath = Utils::buildCategoryPath($breadCrumbs, $navigationCategoryMock);
+        $this->assertSame($expectedCategoryPath, $categoryPath);
+    }
+
+    public function testCategoryPathIsProperlyBuiltWhenMainCategoryIsInADeeperPath(): void
+    {
+        $expectedCategoryPath = 'Cookies_Soft Cookies';
+        $breadCrumbs = ['Main', 'Food', 'Cookies', 'Soft Cookies'];
+
+        $navigationCategoryMock = $this->getMockBuilder(CategoryEntity::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $navigationCategoryMock->expects($this->once())
+            ->method('getBreadcrumb')
+            ->willReturn(['Main', 'Food']);
```

Comment:
So 'Food' is main category, not 'Main'?

Reply:
This are the breadcrumbs. The main category is `Main > Food`.